packages/tpetra/core/src/Tpetra_Details_crsUtils.hpp

```diff
@@ -106,10 +109,13 @@ pad_crs_arrays(
       }
     }, additional_size_needed);
 
-  if (additional_size_needed == 0) return;
+  if (additional_size_needed == 0)
+    return;
 
   // The indices array must be resized and the row_ptr arrays shuffled
   auto indices_new = uninitialized_view<Indices>("ind new", indices.size()+additional_size_needed);
+  auto values_new = uninitialized_view<Values>("val new", pad_values ? values.size()+additional_size_needed : 0);
+  Kokkos::deep_copy(values_new, Teuchos::ScalarTraits<typename Values::value_type>::zero());
```

Comment:
`Kokkos::View` allocation initializes with zero by default; you could save a line of code here by relying on that.